src/shared/sandboxCommandBase.ts

```diff
@@ -0,0 +1,196 @@
+    lifecycle.on(SandboxEvents.EVENT_ASYNC_RESULT, async (results: SandboxProcessObject) => {
+      this.latestSandboxProgressObj = results;
+      this.updateSandboxRequestData();
+      if (!options.isAsync) {
+        this.spinner.stop();
+      }
+      const progress = this.sandboxProgress.getSandboxProgress({ sandboxProcessObj: results, sandboxRes: undefined });
+      const currentStage = progress.status;
+      this.sandboxProgress.markPreviousStagesAsCompleted(currentStage);
+      this.updateStage(currentStage, State.inProgress);
+      this.updateProgress({ sandboxProcessObj: results, sandboxRes: undefined }, options.isAsync);
+      this.log(this.sandboxProgress.formatProgressStatus(false));
+      this.info(
+        messages.getMessage('checkSandboxStatus', [results.Id, options.prodOrg ? options.prodOrg.getUsername() : ''])
+      );
+    });
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    lifecycle.on(SandboxEvents.EVENT_STATUS, async (results: StatusEvent) => {
+      this.latestSandboxProgressObj = results.sandboxProcessObj;
+      this.updateSandboxRequestData();
+      const progress = this.sandboxProgress.getSandboxProgress(results);
+      const currentStage = progress.status;
+      this.updateStage(currentStage, State.inProgress);
+      this.updateProgress(results, options.isAsync);
+    });
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    lifecycle.on(SandboxEvents.EVENT_AUTH, async (results: SandboxUserAuthResponse) => {
+      this.sandboxAuth = results;
+    });
+
+    lifecycle.on(SandboxEvents.EVENT_RESULT, async (results: ResultEvent) => {
+      this.latestSandboxProgressObj = results.sandboxProcessObj;
+      this.updateSandboxRequestData();
+      this.sandboxProgress.markPreviousStagesAsCompleted();
+      this.updateProgress(results, options.isAsync);
+      if (!options.isAsync) {
+        this.progress.stop();
+      }
+      if (results.sandboxRes?.authUserName) {
+        const authInfo = await AuthInfo.create({ username: results.sandboxRes?.authUserName });
+        await authInfo.handleAliasAndDefaultSettings({
+          alias: options.alias,
+          setDefault: options.setDefault,
+          setDefaultDevHub: undefined,
+        });
+      }
+      this.removeSandboxProgressConfig();
+      this.updateProgress(results, options.isAsync);
+      this.reportResults(results);
+    });
+  }
+
+  protected reportResults(results: ResultEvent): void {
+    this.log();
+    this.styledHeader('Sandbox Org Creation Status');
+    this.log(this.sandboxProgress.formatProgressStatus(false));
+    this.logSuccess(
+      [
+        messages.getMessage('sandboxSuccess'),
+        messages.getMessages('sandboxSuccess.actions', [
+          results.sandboxRes?.authUserName,
+          results.sandboxRes?.authUserName,
+        ]),
+      ].join(os.EOL)
+    );
+  }
+
+  protected handleSandboxCreateErrors(prodOrg: Org, err: SfError, isAsync: boolean): SfError {
```

Comment:
this feels weird (return an error to throw).  Why not just throw from `handleSandboxCreateErrors`?

Reply:
Its been removed in latest push